core/src/main/java/org/infinispan/AdvancedCache.java

```diff
@@ -169,6 +169,16 @@
    AuthorizationManager getAuthorizationManager();
 
    /**
+    * Whenever this cache acquires a lock it will do so using the given Object as the owner of said lock.
+    * <p>
+    * This method provides no guarantees as to how this affect locking and should not be used for general usage.
```

Comment:
-100. It's a public API, it's to be used. No hacks. Say what it does, and what it does not (or provide a link to an explanatory document, if you want to keep javadoc concise).